array.go

```diff
@@ -66,14 +81,15 @@ func NewArray(tdbCtx *Context, uri string) (*Array, error) {
 	}
 	curi := C.CString(uri)
 	defer C.free(unsafe.Pointer(curi))
-	array := Array{context: tdbCtx, uri: uri}
-	ret := C.tiledb_array_alloc(array.context.tiledbContext, curi, &array.tiledbArray)
+	var arrayPtr *C.tiledb_array_t
+	ret := C.tiledb_array_alloc(tdbCtx.tiledbContext.Get(), curi, &arrayPtr)
 	if ret != C.TILEDB_OK {
-		return nil, fmt.Errorf("error creating tiledb array: %w", array.context.LastError())
+		return nil, fmt.Errorf("error creating tiledb array: %w", tdbCtx.LastError())
 	}
-	freeOnGC(&array)
+	array := newArrayFromHandle(tdbCtx, newArrayHandle(arrayPtr))
+	array.uri = uri
```

Comment:
Should the URI be set in `newArrayFromHandle` so that an Array created from handles in other places have the URI initialized?

Reply:
`NewArray` is the only case where we've been setting the `uri` field, which is used only by some APIs that should be deprecated either way (#367).